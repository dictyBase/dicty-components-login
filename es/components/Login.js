import React from "react";
import { MuiThemeProvider, createMuiTheme } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import FontAwesome from "react-fontawesome";
import { IconContainer, socialStyles } from "../styles/Buttons";
import "font-awesome/css/font-awesome.min.css";
import "../academicons/css/academicons.min.css";

var generateLoginButtons = function generateLoginButtons(buttons, onClick) {
  return buttons.map(function (name, i) {
    return React.createElement(
      Button,
      {
        style: { backgroundColor: "" + socialStyles[name] },
        onClick: function (_onClick) {
          function onClick() {
            return _onClick.apply(this, arguments);
          }

          onClick.toString = function () {
            return _onClick.toString();
          };

          return onClick;
        }(function () {
          onClick(name);
        }),
        key: i },
      React.createElement(
        IconContainer,
        null,
        name === "orcid" ? React.createElement("i", { className: "ai ai-orcid" }) : React.createElement(FontAwesome, { name: name })
      ),
      "\xA0 Sign in with ",
      name
    );
  });
};

/**
 * This is the main login component that generates the user-specified social buttons with an optional custom theme.
 */

var Login = function Login(_ref) {
  var buttons = _ref.buttons,
      theme = _ref.theme,
      onClick = _ref.onClick;

  // $FlowFixMe
  var newTheme = createMuiTheme(theme);
  return React.createElement(
    MuiThemeProvider,
    { theme: newTheme },
    React.createElement(
      "div",
      null,
      generateLoginButtons(buttons, onClick)
    )
  );
};

export default Login;