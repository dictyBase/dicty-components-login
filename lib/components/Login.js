"use strict";

exports.__esModule = true;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

var _styles = require("@material-ui/core/styles");

var _Button = require("@material-ui/core/Button");

var _Button2 = _interopRequireDefault(_Button);

var _reactFontawesome = require("react-fontawesome");

var _reactFontawesome2 = _interopRequireDefault(_reactFontawesome);

var _Buttons = require("../styles/Buttons");

require("font-awesome/css/font-awesome.min.css");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var generateLoginButtons = function generateLoginButtons(buttons, onClick) {
  return buttons.map(function (name, i) {
    return _react2.default.createElement(
      _Button2.default,
      {
        style: { backgroundColor: "" + _Buttons.socialStyles[name] },
        onClick: function (_onClick) {
          function onClick() {
            return _onClick.apply(this, arguments);
          }

          onClick.toString = function () {
            return _onClick.toString();
          };

          return onClick;
        }(function () {
          onClick(name);
        }),
        key: i },
      _react2.default.createElement(
        _Buttons.IconContainer,
        null,
        name === "orcid" ? _react2.default.createElement("i", { className: "ai ai-orcid" }) : _react2.default.createElement(_reactFontawesome2.default, { name: name })
      ),
      "\xA0 Sign in with ",
      name
    );
  });
};

/**
 * This is the main login component that generates the user-specified social buttons with an optional custom theme.
 */

var Login = function Login(_ref) {
  var buttons = _ref.buttons,
      theme = _ref.theme,
      onClick = _ref.onClick;

  // $FlowFixMe
  var newTheme = (0, _styles.createMuiTheme)(theme);
  return _react2.default.createElement(
    _styles.MuiThemeProvider,
    { theme: newTheme },
    _react2.default.createElement(
      "div",
      null,
      generateLoginButtons(buttons, onClick)
    )
  );
};

exports.default = Login;
module.exports = exports["default"];