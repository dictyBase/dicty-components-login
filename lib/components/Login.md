```js
const buttons = ["orcid", "google", "linkedin", "facebook"]

const theme = {
  overrides: {
    MuiButton: {
      // name of the styleSheet
      root: {
        // name of the rule
        borderRadius: 3,
        color: "white",
        width: "80%",
        justifyContent: "start",
        minHeight: "55px",
        marginBottom: "5px"
      }
    }
  }
}

const handleClick = name => {
  alert(`You clicked ${name}`)
  console.log(`You clicked ${name}`)
}

;<Login buttons={buttons} theme={theme} onClick={handleClick} />
```
