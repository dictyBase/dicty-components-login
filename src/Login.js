// @flow
import React, { Fragment } from "react"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { library } from "@fortawesome/fontawesome-svg-core"
import {
  faGoogle,
  faFacebookF,
  faLinkedinIn,
  faTwitter,
  faYoutube,
  faInstagram,
  faGithubAlt,
  faDropbox,
} from "@fortawesome/free-brands-svg-icons"
import { IconContainer, socialButtonColors, StyledButton } from "./loginStyles"

library.add(
  faGoogle,
  faFacebookF,
  faLinkedinIn,
  faTwitter,
  faYoutube,
  faInstagram,
  faGithubAlt,
  faDropbox,
)

// convert names to desired fontawesome icon name
const name2Icon = {
  google: "google",
  facebook: "facebook-f",
  linkedin: "linkedin-in",
  twitter: "twitter",
  youtube: "youtube",
  instagram: "instagram",
  github: "github-alt",
  dropbox: "dropbox",
}

type Props = {
  /** List of buttons to display */
  buttons: Array<string>,
  /** Custom theme for MuiThemeProvider */
  theme?: Object,
  /** Action to take after user clicks button */
  onClick: Function,
}

/**
 * This is the main login component that generates the user-specified social buttons.
 */

const Login = ({ buttons, onClick }: Props) => (
  <Fragment>
    {buttons.map((name, i) => (
      <StyledButton
        style={{ backgroundColor: `${socialButtonColors[name]}` }}
        onClick={() => {
          onClick(name)
        }}
        key={i}>
        <IconContainer>
          {name === "orcid" ? (
            <i className="ai ai-orcid" />
          ) : (
            <FontAwesomeIcon icon={["fab", name2Icon[name]]} />
          )}
        </IconContainer>
        &nbsp; Sign in with {name}
      </StyledButton>
    ))}
  </Fragment>
)

export default Login
