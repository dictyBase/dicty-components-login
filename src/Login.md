```js
const buttons = ["orcid", "google", "linkedin"]

const handleClick = name => {
  alert(`You clicked ${name}`)
  console.log(`You clicked ${name}`)
}

;<Login buttons={buttons} onClick={handleClick} />
```
