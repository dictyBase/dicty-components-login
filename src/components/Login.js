// @flow
import React from "react"
import { Flex, Box } from "rebass"
import FontAwesomeIcon from "@fortawesome/react-fontawesome"
import { ThemeProvider } from "styled-components"
import {
  FontAwesomeIconContainer,
  DefaultButton,
  theme
} from "../styles/Buttons"
import { Container } from "../styles/Container"
import brands from "@fortawesome/fontawesome-free-brands"

/**
 * This is the main login component that puts all of the individual social buttons in one place.
 */

const loginButtons = buttons => {
  return buttons.map((name, i) => (
    <Flex justify="center" key={i}>
      <Box w={"90%"} mb={"5px"}>
        <DefaultButton size="large" variant="raised" theme={name}>
          <FontAwesomeIconContainer>
            <FontAwesomeIcon icon={["fab", `${name}` ? `${name}` : ""]} />
          </FontAwesomeIconContainer>
          &nbsp; Sign in with {name}
        </DefaultButton>
      </Box>
    </Flex>
  ))
}

const Login = ({ buttons }) => {
  return (
    <ThemeProvider theme={theme}>
      <Container>
        <Flex justify="center">
          <Box w={[1, 1 / 2, 1 / 3]}>{loginButtons(buttons)}</Box>
        </Flex>
      </Container>
    </ThemeProvider>
  )
}

export default Login
