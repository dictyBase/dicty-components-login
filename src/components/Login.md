```js
<Login
  buttons={["google", "facebook", "linkedin", "orcid"]}
  size="large"
  variant="raised"
/>
```
