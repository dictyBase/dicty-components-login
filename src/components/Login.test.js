import React from "react"
import renderer from "react-test-renderer"
import Login from "./Login"
import "jest-styled-components"

test("matching a snapshot of Login", () => {
  const component = renderer.create(<Login />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
