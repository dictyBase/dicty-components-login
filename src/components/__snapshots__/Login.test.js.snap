// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching a snapshot of Login 1`] = `
<div>
  <button
    className="MuiButtonBase-root-16 MuiButton-root-1"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#a6ce39",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      <i
        className="sc-bdVaJa fjJPcf"
      >
        <i
          className="ai ai-orcid"
        />
      </i>
        Sign in with 
      orcid
    </span>
    <span
      className="MuiTouchRipple-root-19"
    />
  </button>
  <button
    className="MuiButtonBase-root-16 MuiButton-root-1"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#dd4b39",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      <i
        className="sc-bdVaJa fjJPcf"
      >
        <span
          aria-hidden={true}
          className="fa fa-google"
        />
      </i>
        Sign in with 
      google
    </span>
    <span
      className="MuiTouchRipple-root-19"
    />
  </button>
  <button
    className="MuiButtonBase-root-16 MuiButton-root-1"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#007bb6",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      <i
        className="sc-bdVaJa fjJPcf"
      >
        <span
          aria-hidden={true}
          className="fa fa-linkedin"
        />
      </i>
        Sign in with 
      linkedin
    </span>
    <span
      className="MuiTouchRipple-root-19"
    />
  </button>
  <button
    className="MuiButtonBase-root-16 MuiButton-root-1"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": "#3b5998",
      }
    }
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-2"
    >
      <i
        className="sc-bdVaJa fjJPcf"
      >
        <span
          aria-hidden={true}
          className="fa fa-facebook"
        />
      </i>
        Sign in with 
      facebook
    </span>
    <span
      className="MuiTouchRipple-root-19"
    />
  </button>
</div>
`;
